format.md

```diff
@@ -131,6 +131,58 @@ is a list of xy vertices. The child name of the outer list should be "polygons";
 the child name of the middle list should be "rings"; the child name of the
 inner list should be "vertices".
 
+**Geometry**: `DenseUnion`
+
+So far, all geometry array types listed above have required that all geometries in the array be of the same type. An array of mixed geometry type is represented as a dense union whose children include one or more of the above geometry array types. Its children may also optionally include a geometry collection array, described below.
+
+The geometry array allows for elements in the array to be of different geometry types.
+
+- The union array may not contain more than one child array of a given geometry type.
+- The "type ids" part of the union field metadata must be defined such that the order of the child arrays match the following order:
+
+    ```
+    1: Point
+    2: LineString
+    3: Polygon
+    4: MultiPoint
+    5: MultiLineString
+    6: MultiPolygon
+    7: GeometryCollection
```

Comment:
I don't understand if such types are also supported?
`GeometryCollection(GeometryCollection(GeometryCollection(MultiPolygon)))`

Reply:
As the PR text stands today, I believe that would be supported, but that's one of the primary aspects that is not yet solidified in this PR. Do you have an opinion on whether it should be supported?

Reply:
If I understand correctly, `List<Geometry>` is a recursive type, and we can't determine the exact number of columns. But in WKB this is valid.

Reply:
I think that's accurate. I'd be open to specifically excluding recursive geometry collections, but I'd want to get consensus on that.

Reply:
I think it's better to rule out recursive geometry collections for now, and get future consensus on _adding_ them, not future consensus on _taking them out_